packages/next/src/build/webpack-config.ts

```diff
@@ -152,7 +152,9 @@ const nodePathList = (process.env.NODE_PATH || '')
 
 const watchOptions = Object.freeze({
   aggregateTimeout: 5,
-  ignored: ['**/.git/**', '**/.next/**'],
+  ignored:
+    // Matches **/node_modules/**, **/.git/** and **/.next/**
+    /^((?:[^/]*(?:\/|$))*)(\.(git|next)|node_modules)(\/((?:[^/]*(?:\/|$))*)(?:$|\/))?/,
```

Comment:
Like this. If this is included in the watcher right now it could slow things down potentially.

<img width="443" alt="image" src="https://github.com/vercel/next.js/assets/14835120/50a37efb-f01b-4604-ab8b-1b4bca749e25">

Reply:
Same setup for us, we have a huge `.yarn/cache` (sitting at 410M right now).

Reply:
Explained this to Alvar already but the `.yarn` folder shouldn't be a problem as it's not resolving files from that directory exactly. It's resolving based on the node_modules / pnp file.

Reply:
Ah, I think I misunderstood you earlier. Though it was checking in another way. All good then.